### PR TITLE
Fix masthead on mobile.

### LINKED ISSUE #2483: /lean mastheads have too much margin on mobile.
image

Removing the margin on the h1 seems to make things look better.

### PR DESCRIPTION
**Pull request by robdodson**
Fixes #2483 

Changes proposed in this pull request:

- Clean up the learning path mastheads so they look better on mobile. Trying to be better about using relative units like `em` to (slowly) nudge us in the right direction for an eventual typography/writespace refactor.

### CODE REVIEW

**Comment by robdodson** [comment]
Failing percy tests are too be expected because I changed the orientation of the image and the text (text first, then image). And the whitespace around the text changed a bit with some padding changes.

Unfortunately percy also lagged when it snapshotted the /about/ page so it look like its blue indicator didn't show up in time :\

File: src/styles/components/_mastheads.scss [hunk 1]
```diff
@@ -75,22 +75,20 @@
 
 .w-masthead-path__container {
   align-items: center;
-  box-sizing: content-box;
   display: flex;
-  flex-direction: column-reverse;
```
Comment (samthor) [inline src/styles/components/_mastheads.scss:80]: did `-reverse` actually manifest in a serious way before?

Comment (robdodson) [reply src/styles/components/_mastheads.scss:80]: > did -reverse actually manifest in a serious way before?

it used to make the image go before the text on mobile. That was the original design but I changed it in this PR

**Comment by robdodson** [comment]
pushing an empty commit to see if i can make that one flaky percy snapshot do the right thing

**Pull request merged**